packages/camera/camera_windows/windows/capture_engine_listener.h

```diff
@@ -0,0 +1,62 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef PACKAGES_CAMERA_CAMERA_WINDOWS_WINDOWS_CAPTURE_ENGINE_LISTENER_H_
+#define PACKAGES_CAMERA_CAMERA_WINDOWS_WINDOWS_CAPTURE_ENGINE_LISTENER_H_
+
+#include <mfcaptureengine.h>
+
+#include <functional>
+
+namespace camera_windows {
+
+class CaptureEngineObserver {
+ public:
+  virtual ~CaptureEngineObserver() = default;
+
+  // Returns true if sample can be processed.
+  virtual bool IsReadyForSample() = 0;
+
+  // Handles Capture Engine media events.
+  virtual void OnEvent(IMFMediaEvent* event) = 0;
+
+  // Updates texture buffer
+  virtual bool UpdateBuffer(uint8_t* data, uint32_t new_length) = 0;
+
+  // Handles capture timestamps updates.
+  // Used to stop timed recordings when recorded time is exceeded.
+  virtual void UpdateCaptureTime(uint64_t capture_time) = 0;
+};
+
+class CaptureEngineListener : public IMFCaptureEngineOnSampleCallback,
+                              public IMFCaptureEngineOnEventCallback {
```

Comment:
Add class documentation. Since it looks like null observers are allowable, it's definitely worth touching on that usecase in the doc comment.